sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -413,16 +423,24 @@ def all_params(parameters):
     # transform default values
     for left_args in parameters:
         if left_args.name not in transformed_kwargs.keys():
-            default = left_args.default if left_args.default is not Parameter.empty else None
-            transformed_kwargs[left_args.name] = _wrap_pipeline_parameter(left_args.name, default)
+            default_value = left_args.default if left_args.default is not Parameter.empty else None
+            actual_value = user_provided_kwargs.get(left_args.name)
+            transformed_kwargs[left_args.name] = _wrap_pipeline_parameter(
+                key=left_args.name, default_value=default_value, actual_value=actual_value
+            )
     return transformed_kwargs
 
 
-def _wrap_pipeline_parameter(key, value, group_names=None):
+def _wrap_pipeline_parameter(key, default_value, actual_value, group_names=None):
```

Comment:
what's the difference between default_value & actual value?

Reply:
For example,

```
@pipeline 
def my_pipeline(int_param=1):
    pass

my_pipeline(int_param=2)
```

`1` will be default value(I think this should appear in pipeline component, but we did not use this currently), `2` will be actual value.